src/main/java/io/curity/identityserver/plugin/OptInMFA/AuthenticatorModel.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright 2020 Curity AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.curity.identityserver.plugin.OptInMFA;
+
+import se.curity.identityserver.sdk.Nullable;
+import se.curity.identityserver.sdk.service.authenticationaction.AuthenticatorDescriptor;
+
+public final class AuthenticatorModel
+{
+    private final String _acr;
+    private final String _description;
+    private final String _type;
+
+    public AuthenticatorModel(String acr, @Nullable String description, String type, String name)
+    {
+        _acr = acr;
+
+        if (!name.equals(acr)) {
+            _description = name;
+        }
+        else if (description != null)
+        {
+            _description = description;
```

Comment:
Why is this only one if `name != acr`?
I think this would make more sense:

```java
_description = description != null ? description : name;
```

Reply:
The idea is that `name` will be set by the user when registering their authenticator. But if they don't set any name then the name will be equal `acr`. When displaying the list I want to first use the name set by the user, then the description from `acr`, and only then the `acr` value. So the first if - `!name.equals(acr)` - is done to differentiate from user set name and the default name.

Reply:
But that still doesn't seem right to me. `_description` is not assigned the value of `description` when `!name.equals(acr)` even if `_description` is not null.

> But if they don't set any name then the name will be equal acr

Why does that affect `description` though?

> the first if - !name.equals(acr) - is done to differentiate from user set name and the default name.

This is impossible to know by anyone else by reading the code. I would suggest doing this to make this assumption clear:
```java
boolean isCustomName = !name.equals(acr);
```
Now, without other changes, the code would look like this:

```java
_description = isCustomName ? name : description;
```
Which is very surprising :) I think you are using `description` and `name` for the same purpose, but that's unclear exactly why.

Reply:
I will introduce the `isCustomName` variable. That will make it definitely more clear there.

So there is a little bit of confusion in the property / variable names. 

In the `AuthenticatorDescriptor` class, the field `description` is actually the label, used when displaying the list of available authenticators. That's why I called mine field `description`. But I want it to have the user's label, which the value the user chose for the name of their authenticator.

I can call this field `label` and then there will be four different variables used ;) So the `label` would have the value of either: `name` (which is the one set by the user), `description` (which is the label from the authenticator) or `acr` (which is the ACR value, just in case if the description is not set for a given authenticator).